exercises/circular-buffer/example.jl

```diff
@@ -0,0 +1,177 @@
+module CircularBuffers
+
+using Base: @propagate_inbounds
+using Printf
+
+export CircularBuffer, capacity, isfull
+
+const OVERWRITE = false
```

Comment:
Every few minutes, because I was switching between the generated tests and the tests from DataStructures.jl where overwriting was the default. :smile: But you're right, I should've taken this out afterward and I've made the edit.